### PR TITLE
Add components and UI for individual conditions

### PR DESCRIPTION
**Pull request by benswannack**
## Context
There are new components required for the individual conditions flow

## Changes proposed in this pull request
- Add a conditions list component with the correct HTML from the. prototype
- Add a form page for selecting condition statuses
- Add a check answers page

## Guidance to review
Per commit

## Link to Trello card
https://trello.com/c/4keyF4am/3791-implement-individual-conditions-ui-in-manage-for-non-deferred-offers

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/components/provider_interface/individual_condition_status_review_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<span class="govuk-caption-l"><%= application_choice.application_form.full_name %></span>
+<h1 class="govuk-heading-l"><%= page_heading %></h1>
+
+<%= render ProviderInterface::OfferSummaryComponent.new(
+  application_choice: application_choice,
+  course_option: application_choice.current_course_option,
+  conditions: form_object.conditions,
+  course: application_choice.current_course,
+) %>
```
Comment (benswannack) [inline app/components/provider_interface/individual_condition_status_review_component.html.erb:9]: There's a bug with this at the moment where if the course has multiple study modes, then a change link renders. Need to modify the base component to fix this

Comment (benswannack) [reply app/components/provider_interface/individual_condition_status_review_component.html.erb:9]: Resolved by extending the `editable` attribute of the offer summary component

File: app/components/provider_interface/conditions_list_component.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module ProviderInterface
+  class ConditionsListComponent < ViewComponent::Base
+    attr_reader :conditions
+
+    def initialize(conditions)
+      @conditions = conditions || []
```
Comment (despo) [inline app/components/provider_interface/conditions_list_component.rb:6]: The association should always return an empty array if it's not set so I don't think this is needed.

```suggestion
      @conditions = conditions 
```

File: app/components/provider_interface/individual_condition_status_review_component.html.erb (hunk 1 above)
Comment (despo) [reply app/components/provider_interface/individual_condition_status_review_component.html.erb:9]: Can the ReadOnlyCompletedOfferSummaryComponent not be used instead?

Comment (benswannack) [reply app/components/provider_interface/individual_condition_status_review_component.html.erb:9]: Oh I didn't realise that was a thing! What is the intended difference between `editable: false` and the read only component?

Comment (despo) [reply app/components/provider_interface/individual_condition_status_review_component.html.erb:9]: I don't remember the details but editable was introduced in order to be used in conjuction with the ReadOnly view. The read only view does something similar to what you've done in OfferSummary, by updating `rows` to not include any links.

https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/views/provider_interface/offers/show.html.erb#L27

Comment (benswannack) [reply app/components/provider_interface/individual_condition_status_review_component.html.erb:9]: Ah yes, what I've done won't work. `editable: false` also disables editing of the conditions, which I do need here.

I'll use the readonly one here and maybe we can revisit the components another time if necessary

**Review by cwrw — APPROVED** [review]
Looks good 👍  past addressing the other comment

**Pull request merged**